pkg/provisioning/parse.go

```diff
@@ -44,6 +44,14 @@ type PipelineConfig struct {
 	Description string                     `yaml:"description"`
 	Connectors  map[string]ConnectorConfig `yaml:"connectors,omitempty"`
 	Processors  map[string]ProcessorConfig `yaml:"processors,omitempty"`
+	DLQ         DLQConfig                  `yaml:"dead-letter-queue"`
+}
+
+type DLQConfig struct {
+	Plugin              string            `yaml:"plugin"`
+	Settings            map[string]string `yaml:"settings"`
+	WindowSize          *int              `yaml:"window-size"`
```

Comment:
is there a specific reason to have it as a pointer?

Reply:
Yes, setting the window size to 0 is a valid thing to do, while if it's not set the default window size is 1, so we have to be able to differentiate if the window size was explicitly set to 0 or if it is empty.

Reply:
oh makes sense! thanks for the explanation